### PR TITLE
Plugins: ES6ify the PluginActivateToggle component

### PR DESCRIPTION
**Pull request by tyxla**
This PR ES6ifies the `PluginActivateToggle` component, converting it into an ES6 class, and fixing all warnings. It also converts the `lib/analytics` usage to the Redux `state/analytics`, and fixes the tests.

To test:

* Checkout this branch
* Go to a page that has a list of plugins (like `http://calypso.localhost:3000/plugins/$site`, where `$site` is one of your Jetpack sites).
* Play around with activation and deactivation of plugins and verify there are no regressions with the display and functionality of the activation toggle.
* Verify the component tests still pass: `npm run test-client client/my-sites/plugins/plugin-activate-toggle/`.

/cc @johnHackworth @ryelle @oskosk

### CODE REVIEW

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx [hunk 1]
```diff
@@ -1,87 +1,95 @@
...
-				htmlFor={ 'activate-' + this.props.plugin.slug + '-' + this.props.site.ID }
+				htmlFor={ 'activate-' + plugin.slug + '-' + site.ID }
 				/>
 		);
 	}
-} );
+}
+
+PluginActivateToggle.propTypes = {
+	site: PropTypes.object.isRequired,
```
Comment (gwwar) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: >  Not sure if this is a good idea,

It is 😄 but I can understand if there's too much to untangle at once. Usually, it's a bit easier if we update child components to handle their own data needs first. @tyxla feel free to make a judgement call on if you want to clean up the site passing here or in a follow up PR.

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx [hunk 2]
```diff
@@ -1,87 +1,95 @@
...
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Activate Plugin', 'Plugin Name', plugin.slug );
+			this.props.recordTracksEvent( 'calypso_plugin_activate_click', {
+				site: site.ID,
+				plugin: plugin.slug
 			} );
 		}
-	},
+	}
```
Comment (obenland) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: I couldn't find terribly useful docs on this, but it looks like class properties still need a closing semicolon

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx [hunk 3]
```diff
@@ -1,87 +1,95 @@
...
+		PluginsActions.togglePluginActivation( site, plugin );
+		PluginsActions.removePluginsNotices( notices.completed.concat( notices.errors ) );
 
-		if ( this.props.plugin.active ) {
-			analytics.ga.recordEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', this.props.plugin.slug );
-			analytics.tracks.recordEvent( 'calypso_plugin_deactivate_click', {
-				site: this.props.site.ID,
-				plugin: this.props.plugin.slug
+		if ( plugin.active ) {
+			this.props.recordGoogleEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
```
Comment (gwwar) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: We can also create a new variable name if you get tired of typing this out:
```javascript
const { recordGoogleEvent: recordGAEvent } = this.props;
recordGAEvent( 'Plugins', 'Clicked Toggle Deactivate Plugin', 'Plugin Name', plugin.slug );
```

**Comment by johnHackworth** [comment]
@enejb needs to check this and see that all the cases works right before merging

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx (hunk 1 above)
Comment (tyxla) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: Let's do that in another PR. You folks are right, but I have bad memories from doing multipurpose PRs.

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx [hunk 4]
```diff
@@ -1,87 +1,95 @@
...
 		}
 
-		inProgress = PluginsLog.isInProgressAction( this.props.site.ID, this.props.plugin.slug, [
+		const inProgress = PluginsLog.isInProgressAction( site.ID, plugin.slug, [
 			'ACTIVATE_PLUGIN',
 			'DEACTIVATE_PLUGIN'
 		] );
 
-		if ( this.props.plugin && 'jetpack' === this.props.plugin.slug ) {
+		if ( plugin && 'jetpack' === plugin.slug ) {
```
Comment (tyxla) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: I'd vote for keeping this check in case a `null` value gets passed.

**Comment by tyxla** [comment]
Thanks for the feedback, @gwwar @obenland! I've addressed your comments, and fixed an issue with importing the component within another components. I haven't ES6-ified them, as that would go outside of the PR scope.

Looking forward to some feedback from @enejb as well.

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx (hunk 4 above)
Comment (obenland) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: Is that a possibility? 
I'm just wondering what good an activate toggle is if there's no plugin.

**Comment by enejb** [comment]
This works as expected. I tested the toggle in the few different places that it does show up. 
In my testing I found a few different regressions that I will add as issues that should be fixed in a different PR. 

Thanks doing this much needed changes @tyxla. The code is much more readable as a result. 
I think this is ready to :ship:

File: client/my-sites/plugins/plugin-activate-toggle/index.jsx (hunk 4 above)
Comment (tyxla) [inline client/my-sites/plugins/plugin-activate-toggle/index.jsx]: After doing some tests here and there, I confirm that this is not happening. However, I'm still sticking to my opinion here, because it's possible that such cases might occur while we're migrating plugins to Redux. We've seen that happening with the `getSelectedSite()` migration, for example in #9562.
